docs/src/main/paradox/healthchecks.md

```diff
@@ -0,0 +1,68 @@
+# Heath checks
+
+Akka Management supports two kinds of health checks:
+
+* Readiness checks: should the application receive external traffic, for example waiting for the following to complete
+    * Joining a Cluster
+    * Establishing a connection to a database or queuing system
+* Liveness checks: should the application be left running 
+
+Readiness checks can be used to decide if a load balancer should route traffic where
+as liveness checks can be used in environments which can restart a hung process.
+
+This matches [Kubernetes Health checks](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-probes/).
+
+## Defining a Health Check
+
+A health check is conceptually of the shape @scala[`ActorSystem => Future[Boolean]`]@java[`ActorSystem => CompletionStage<Boolean>`].
```

Comment:
Will Java users understand the arrow function type syntax? Maybe better to write it out?

```suggestion
A health check is conceptually a function from an `ActorSystem` to a @scala[`Future[Boolean]`]@java[`CompletionStage<Boolean>`].
```

Reply:
Actually, now that I read below that these actually take the `ActorSystem` in a constructor, I wonder if this sentence just confuses things. Maybe it's better to drop this sentence, and move "A health check must extend..." up here.

Reply:
Agreed!

Reply:
also, I think we moved the ActorSystem to a constructor parameter
(and Java syntax would be `->` if any arrow)